examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -320,24 +321,81 @@ CastingServer::ContentLauncherLaunchURL(TargetEndpointInfo * endpoint, const cha
 
 void CastingServer::DeviceEventCallback(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
 {
+    bool runPostCommissioning           = false;
+    chip::NodeId targetPeerNodeId       = 0;
+    chip::FabricIndex targetFabricIndex = 0;
     if (event->Type == DeviceLayer::DeviceEventType::kBindingsChangedViaCluster)
     {
+        ChipLogProgress(AppServer, "CastingServer::DeviceEventCallback kBindingsChangedViaCluster received");
         if (CastingServer::GetInstance()->GetActiveTargetVideoPlayer()->IsInitialized())
         {
+            ChipLogProgress(AppServer,
+                            "CastingServer::DeviceEventCallback already connected to video player, reading server clusters");
             CastingServer::GetInstance()->ReadServerClustersForNode(
                 CastingServer::GetInstance()->GetActiveTargetVideoPlayer()->GetNodeId());
         }
+        else if (CastingServer::GetInstance()->udcInProgress)
+        {
+            ChipLogProgress(AppServer,
+                            "CastingServer::DeviceEventCallback UDC is in progress while handling kBindingsChangedViaCluster");
+            CastingServer::GetInstance()->udcInProgress = false;
+            if (CastingServer::GetInstance()->mTargetVideoPlayerNumIPs > 0)
+            {
+                TargetVideoPlayerInfo * connectableVideoPlayerList =
+                    CastingServer::GetInstance()->ReadCachedTargetVideoPlayerInfos();
+                if (connectableVideoPlayerList == nullptr || !connectableVideoPlayerList[0].IsInitialized())
+                {
+                    ChipLogProgress(AppServer, "No cached video players found");
+                }
+
+                for (size_t i = 0; i < kMaxCachedVideoPlayers && connectableVideoPlayerList[i].IsInitialized(); i++)
+                {
+                    if (connectableVideoPlayerList[i].IsSameAs(CastingServer::GetInstance()->mTargetVideoPlayerDeviceName,
+                                                               CastingServer::GetInstance()->mTargetVideoPlayerNumIPs,
+                                                               CastingServer::GetInstance()->mTargetVideoPlayerIpAddress))
+                    {
+                        ChipLogProgress(AppServer,
+                                        "CastingServer::DeviceEventCallback found the video player to initialize/connect to");
+                        targetPeerNodeId     = connectableVideoPlayerList[i].GetNodeId();
+                        targetFabricIndex    = connectableVideoPlayerList[i].GetFabricIndex();
+                        runPostCommissioning = true;
+                    }
+                }
```

Comment:
maybe log a message if we didn't find a matching connectableVideoPlayer - this would be helpful for debugging I think

Reply:
Done. Logging an error message, calling back the client with an error and returning early.